src/resolve/resolved-manager.c

```diff
@@ -1929,3 +1930,112 @@ void dns_manager_reset_statistics(Manager *m) {
         m->n_failure_responses_served_stale_total = 0;
         zero(m->n_dnssec_verdict);
 }
+
+static int dns_configuration_json(
+                const char *ifname,
+                int ifindex,
+                DnsServer *current_dns_server,
+                DnsServer *dns_servers,
+                DnsSearchDomain *search_domains,
+                sd_json_variant **ret) {
+
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *dns_servers_json = NULL,
+                                                          *search_domains_json = NULL;
+        int r;
+
+        assert(ret);
+
+        r = sd_json_variant_new_array(&dns_servers_json, NULL, 0);
+        if (r < 0)
+                return r;
+
+        r = sd_json_variant_new_array(&search_domains_json, NULL, 0);
+        if (r < 0)
+                return r;
+
+        LIST_FOREACH(servers, s, dns_servers) {
+                r = sd_json_variant_append_arrayb(&dns_servers_json, SD_JSON_BUILD_STRING(dns_server_string_full(s)));
```

Comment:
By reconsidering about this, I think it is better to build structured information for DNS server. That is, store address (byte array), port (unsigned), interface specifier (integer), server name (string). Like resolved already does in DBus and other varlink methods.

Reply:
yes, please generate a proper json representation of the server fields.

Reply:
Thanks, I have done this in the latest push.